h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -539,56 +568,41 @@ private void addGridKey(Key<Grid> gridKey) {
   /**
    * Do a random hyperparameter search.  Caller must eventually do a <i>get()</i>
    * on the returned Job to ensure that it's complete.
-   * @param algoName name of the algo, e.g. "GBM"; used for messages and for building the grid key
+   * @param algo the Algo, e.g. {@link Algo#GBM}; used for validation, messages and for building the grid key
    * @param baseParms ModelBuilder parameter values that are common across all models in the search
    * @param searchParms hyperparameter search space
    * @return the started hyperparameter search job
    */
-  public Job<Grid> hyperparameterSearch(String algoName, Model.Parameters baseParms, Map<String, Object[]> searchParms) {
-    return hyperparameterSearch(null, algoName, baseParms, searchParms);
+  public Job<Grid> hyperparameterSearch(Algo algo, Model.Parameters baseParms, Map<String, Object[]> searchParms) {
+    return hyperparameterSearch(null, algo, baseParms, searchParms);
   }
 
   /**
    * Do a random hyperparameter search.  Caller must eventually do a <i>get()</i>
    * on the returned Job to ensure that it's complete.
    * @param gridKey optional grid key
-   * @param algoName name of the algo, e.g. "GBM"; used for messages and for building the grid key if it's not specified
+   * @param algo  the Algo, e.g. "GBM"; used for messages and for building the grid key if it's not specified
    * @param baseParms ModelBuilder parameter values that are common across all models in the search
    * @param searchParms hyperparameter search space
    * @return the started hyperparameter search job
    */
-  public Job<Grid> hyperparameterSearch(Key<Grid> gridKey, String algoName, Model.Parameters baseParms, Map<String, Object[]> searchParms) {
-    if (ArrayUtils.contains(skipAlgosList, algoName)) {
-      userFeedback.info(Stage.ModelTraining,"AutoML: skipping algo " + algoName + " hyperparameter search");
-      return null;
-    }
+  public Job<Grid> hyperparameterSearch(Key<Grid> gridKey, Algo algo, Model.Parameters baseParms, Map<String, Object[]> searchParms) {
+    if (exceededSearchLimits(algo, JobType.HyperparamSearch)) return null;
```

Comment:
Is there a reason why the `UserFeedback` is not needed?

Reply:
closed